File: Frontend/src/app/shared/models/employee.ts
import { PositionType } from './../enums/position-type';
export class Employee {
  id: number;
  firstName: string;
  lastName: string;
  birthDate: Date;
  positionType: PositionType;
  offerStatueses: JobOfferStatus[];

  constructor(id: number, firstName: string, lastName: string, birthDate: Date, positionType: PositionType, offerStatueses: JobOfferStatus[]){
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.birthDate = birthDate;
    this.positionType = positionType;
    this.offerStatueses = offerStatueses;
  }
}

File: Backend/EventManager.Domain/Models/Employee.cs
using EventManager.Domain.Enums;
using System;
using System.Collections.Generic;

namespace EventManager.Domain.Models
{
    public class Employee
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public PositionType PositionType { get; set; }
        public ICollection<JobOfferStatus> OfferStatuses { get; set; }

        public void AddStatus(JobOfferStatus status)
        {
            if (OfferStatuses is null)
            {
                OfferStatuses = new List<JobOfferStatus>() { status };
            }
            else
            {
                OfferStatuses.Add(status);
            }
        }
    }
}
